pkg/identityserver/user_registry.go

```diff
@@ -248,16 +247,6 @@ func (is *IdentityServer) createUser(ctx context.Context, req *ttnpb.CreateUserR
 	defer func() { is.setFullProfilePictureURL(ctx, usr) }()
 
 	err = is.store.Transact(ctx, func(ctx context.Context, st store.Store) (err error) {
-		if req.InvitationToken != "" {
```

Comment:
Why shouldn't we keep the ordering here ? You are creating the user even if the invitation is expired. 

I know that the transaction will fail but I don't see why it shouldn't fail early.

Reply:
I took the transaction for granted and decided to eliminate the duplicate check in the beginning since the `SetInvitationAcceptedBy` used the `usr` response from the `CreateUser`.

Changed so that the validation happens before the user creation.